packages/nimble-components/src/theme-provider/index.ts

```diff
@@ -33,19 +53,46 @@ export const theme = DesignToken.create<Theme>({
  * @internal
  */
 export class ThemeProvider extends FoundationElement {
-    @attr({
-        attribute: 'direction'
-    })
+    @attr()
+    public override lang = '';
```

Comment:
I'll comment because I'm still thinking about it. A concern I have about overriding the lang attribute is that if we override the lang attribute we should match the attributes behavior identically. We shouldn't override an attribute and not align with the browser behavior.

My concern is that this does not behave identically. Setting to empty string will reflect that value to the dom. I don't think `<div lang>` and `<div>` are identical in terms of how they impact lang settings. Not sure but also not desirable attribute manipulation behavior, we should minimize changing the behavior of attributes that we don't own and setting the attr to empty string is a change in behavior to normal elements, i.e. any newly created `div` element won't magically get a `lang` attribute added to it.

Reply:
We could leave `lang` without an initializer and just suppress the typescript error about the missing initializer by using a non-null assertion. I realize that's a lie, but if we have a comment that explains why we're lying, maybe that's a reasonable solution.

Reply:
Is there a reason not to give this attribute a different name (e.g. `language` or `locale`) rather than trying to force it to align with the browser behavior?

Reply:
Might be confusing to have two separate attributes that both effectively specify the locale to use for the contents. I could see someone using `lang` on accident when they should have used `language`.

Reply:
My concern is the DOM state getting out of sync with token state:

```html
<html lang="en">
<style>
div:lang(en) {color: red;}
div:lang(fr) {color: blue;}
</style>
<nimble-theme-provider language="fr">
    <div>Hello!</div>
</nimble-theme-provider>
</html>
```

https://jsbin.com/luvomuqosu/edit?html,js,output

To align the DOM state you have to do something like:

```html
<html lang="en">
<style>
div:lang(en) {color: red;}
div:lang(fr) {color: blue;}
</style>
<nimble-theme-provider lang="fr" language="fr">
    <div>Hello!</div>
</nimble-theme-provider>
</html>
```

https://jsbin.com/yupayavuza/edit?html,js,output

But if we could use the native lang then the DOM state is aligned:

```html
<html lang="en">
<style>
div:lang(en) {color: red;}
div:lang(fr) {color: blue;}
</style>
<nimble-theme-provider lang="fr">
    <div>Hello!</div>
</nimble-theme-provider>
</html>
```

https://jsbin.com/vaveyameju/edit?html,js,output


Our usage guidelines then become something like whenever you need to update the locale for a region of the page you have to use the theme provider. Don't set it on a random div, wrap the content in a theme provider and set `lang` and then the token state and the actual dom state align

Reply:
Discussed in post-stand-up:
1. No pushback on using the same attribute so that the DOM configuration and token configuration are controlled via a single property
2. The semantics of the lang attribute + token need to align with the DOM
   - If unset resolves to parent value?
   - If set to invalid value, resolve to parent?
   - If set to empty string (mdn describes as the "unknown" state) what is the behavior? Resolved to parent?
   - If the root (html element lang attr) is unset, resolve to system value? Is that the same as DateFormat(undefined) locale?
   - If the root is set to invalid value, resolve to system?
   - If the root is set to empty string, resolve to system?

Some research needed there, so moving to draft as per new contributing policy. Can `@mention` for discussions as researched and peer review, etc.

Reply:
>If unset resolves to parent value?

Yes. But in the case of the token, we can't watch every ancestor element, so we only look at the root. If lang is unset on the theme provider, we fall back to the root's lang. If the root's lang is unset, the lang is _unknown_, equivalent to a value of empty string.

>If set to invalid value, resolve to parent?

No. The value remains that specific invalid value. It is up to consumers of the locale to determine how to handle an invalid value.

>If set to empty string (mdn describes as the "unknown" state) what is the behavior? Resolved to parent?

No, remains empty string.

>If the root (html element lang attr) is unset, resolve to system value? Is that the same as DateFormat(undefined) locale?

No, root unset just means _unknown_ locale. The `DateTimeFormat()` constructor resolves `undefined` to the browser's language, but empty string (unknown) and invalid values result in a thrown error.

>If the root is set to invalid value, resolve to system?

No, remains invalid value.

>If the root is set to empty string, resolve to system?

No, means the same as unset: _unknown_.

So...
1. our token may have an invalid/empty value. 
2. we may fail to create a date-time formatter with the token value
3. should we have a fall-back locale to create our formatter with when the token value is bad? Use the browser locale?

Reply:
> Yes. But in the case of the token, we can't watch every ancestor element, so we only look at the root. If lang is unset on the theme provider, we fall back to the root's lang. If the root's lang is unset, the lang is unknown, equivalent to a value of empty string.

Nope. We just discussed this. It's not the root but theme-providers in the cascade. Theme providers can be nested. Should look to the cascade (which is just how DesignTokens work, this is nothing new).

```html
<html lang="fr">
  <theme-provider lang="en">
    <theme-provider>
        <!-- Token lang value and DOM lang value is en, not fr -->
    </theme-prover>
  </theme-provier>
</html>
```

> our token may have an invalid/empty value.

The theme-provider lang attribute can be configured to an invalid value. Whether we use that same value on the token is the question. Need to understand the trade-offs of letting the token value be invalid to align with DOM and making every component responsible for invalid values or choosing to differ from the DOM and having the token value always resolve to a valid value.

Reply:
What research was done to determine the default locale? 

Why was the DateFormatter(undefined) option chosen as the default over navigator.language? 

https://stackoverflow.com/questions/57666095/browser-default-locale-intl-datetimeformat-vs-navigator-language

Reply:
> Nope. We just discussed this. It's not the root but theme-providers in the cascade and then the root.

I was just describing the single theme provider case. My understanding is that we don't have to do anything special to handle the nested theme-provider case. We're just concerned with what the token defaults to, because that is what applies when the theme provider cascade provides no `lang` value.

> Need to understand the trade-offs...

If we allow our token to have an invalid/unknown value, there are a couple primary drawbacks:
- any code using the token value must be aware of and add logic to handle an invalid/unknown value
- different consumers of the token may handle invalid/unknown values inconsistently, i.e. not falling back to another locale, or falling back to a different one

The only advantage of allowing invalid/unknown values is:
- resulting behavior/errors may make it easier to detect and fix a bad value that was specified by accident

Based on these points, I would advocate for preventing invalid/unknown locale values in our `lang` token.

> What research was done to determine what we should use as the default locale?
> Why was the DateFormatter(undefined) option chosen as the default over navigator.language? 

I thought following the precedent of `Intl.DateTimeFormat`'s handling of an unspecified locale was a good approach. I didn't know about `navigator.language`. It appears that `Intl.DateTimeFormat` falls back to the system locale, whereas `navigator.language` uses the browser's language setting. Both seem reasonable to me, but maybe we should prefer the latter, since it is more tightly scoped.

Reply:
We changed our decision to fall back to "en-US" instead of navigator.language.

Where should we document this? We don't have a Storybook story for the theme provider, but maybe we should make one? Don't know what it would render, though.

Reply:
Can follow the label-provider doc patterns, see @msmithNI

Reply:
I don't know how that would translate to this `lang` token documentation. For label-provider, we created a "Label Providers" Docs page under the "Tokens" section. It seems like the analog would be a "Theme Provider" Docs page, but I question the discoverability of a "lang" token under "Theme Provider" docs. Since the section is "Tokens", should we just give the `lang` token its own docs page as a sibling of "Base Colors", "Theme-aware Tokens", and "Label Providers"? If it helps, we could discuss this over Teams.

Reply:
Think the discoverability of label provider under tokens is questionable for discoverability as well so I don't think your alternative proposals provide significant advantage.

I'd suggest making theme-provider next to label-providers for now and it can document its theme, direction, and lang tokens. We can revisit how we document them in storybook in the future.

If that is not palatable now, I don't have the bandwidth for this topic this week. Can brainstorm with another team member.

Reply:
Added mdx for theme provider with documentation for theme, lang, and direction.